sound/soc/sof/pcm.c

```diff
@@ -325,6 +325,42 @@ static int sof_pcm_trigger(struct snd_pcm_substream *substream, int cmd)
 	/* send IPC to the DSP */
 	ret = sof_ipc_tx_message(sdev->ipc, stream.hdr.cmd, &stream,
 				 sizeof(stream), &reply, sizeof(reply));
+	if (ret < 0)
+		return ret;
+
+	/* do we need to inform runtime PM we can enter D3 ? */
+	permit_d0i3 = snd_sof_pipeline_permit_d0i3(sdev,
+						   spcm->swidget->pipeline_id);
+	if (!permit_d0i3)
+		return 0;
+
+	/* inform runtime PM driver can enter D3 */
```

Comment:
@plbossart @lgirdwood I'm a bit confused about this part. We will not even get a suspend trigger for streams associated with dai_links whose .ignore_suspend is set. This should help us determine when to activate D3.